File: src/main/java/vocadb/notification/reader/client/query/NameMatchMode.java
package vocadb.notification.reader.client.query;

import com.fasterxml.jackson.annotation.JsonProperty;

public enum NameMatchMode {
    @JsonProperty("None")
    AUTO,
    @JsonProperty("Partial")
    PARTIAL,
    @JsonProperty("StartsWidth")
    STARTS_WIDTH,
    @JsonProperty("Exact")
    EXACT,
    @JsonProperty("Words")
    WORDS;
}

File: src/main/java/vocadb/notification/reader/client/api/enums/NameMatchMode.kt
package vocadb.notification.reader.client.api.enums

import com.fasterxml.jackson.annotation.JsonValue

enum class NameMatchMode(@JsonValue val value: String) {
    AUTO("None"),
    PARTIAL("Partial"),
    STARTS_WIDTH("StartsWidth"),
    EXACT("Exact"),
    WORDS("Words");
}
